// matching.proto

syntax = "proto3";

package dgdo.matching;

import "common.proto";

// --------------------
// Candidate driver
// --------------------
message Candidate {
  string driver_id = 1;
  double probability = 2;           // normalized, sum â‰ˆ 1
  double distance_meters = 3;       // optional, for ranking
  int32 eta_seconds = 4;            // optional
}

// --------------------
// Matching request (command)
// --------------------
message MatchingRequest {
  string trip_request_id = 1;         // immutable reference / idempotency key

  dgdo.common.Location origin = 2;
  dgdo.common.Location destination = 3;

  uint32 max_candidates = 4;          // hard cap
  int64 seed = 5;                     // deterministic replay

  dgdo.common.Metadata metadata = 6;  // optional, for ML/telemetry
}

// --------------------
// Matching response
// --------------------
message MatchingResponse {
  repeated Candidate candidates = 1;
  string reason_code = 2;             // e.g., "NO_DRIVERS", empty if OK
}

// --------------------
// Service
// --------------------
service MatchingService {
  rpc GetCandidates(MatchingRequest) returns (MatchingResponse);
}
